### ISSUE TITLE
Bug: -prefix switches to absolute paths instead of relative

### ISSUE THREAD

**Issue by ghostsquad** [shuLhan/go-bindata#22]
## Expected Behavior

When using `-prefix`, paths continue to be relative, if they were passed in as relative.

## Actual Behavior

Paths get changed to absolute, and weird munging happens (middle of the path gets removed/nullified)

## Repro

```
go-bindata  -o ./examples/basic/lib/bindata.go ./examples/basic/docs/docopt.txt
```
I see in `bindata.go`

```
var _bintree = &bintree{nil, map[string]*bintree{
	"examples": {nil, map[string]*bintree{
		"basic": {nil, map[string]*bintree{
			"docs": {nil, map[string]*bintree{
				"docopt.txt": {examplesBasicDocsDocoptTxt, map[string]*bintree{}},
			}},
		}},
	}},
}}
```

but when I pass in `--prefix` like:
```
go-bindata --prefix examples/basic  -o ./examples/basic/lib/bindata.go ./examples/basic/docs/docopt.txt
```

I get:
```
var _bintree = &bintree{nil, map[string]*bintree{
	"Users": {nil, map[string]*bintree{
		"wes": {nil, map[string]*bintree{
			"projects": {nil, map[string]*bintree{
				"ronn2docopt": {nil, map[string]*bintree{
					"": {nil, map[string]*bintree{
						"docs": {nil, map[string]*bintree{
							"docopt.txt": {usersWesProjectsRonn2docoptDocsDocoptTxt, map[string]*bintree{}},
						}},
					}},
				}},
			}},
		}},
	}},
}}
```

I think this is a bug. If the prefix never contained an absolute path before trying to strip it out, why would it start using an absolute path.

## Workaround
```
-prefix "$(pwd)/examples/basic"
```

**Comment by shuLhan** [comment]
@ghostsquad 

Thank you for the report. Since the original code was not written by me, I am in the middle of figuring it out how the internal process of mapping of input to output work in go-bindata. There will be major refactoring later but It's take time (free time on weekend only).

In the meantime, you can use the workaround or use regular expression as noted in readme [1].

[1] https://github.com/shuLhan/go-bindata#path-prefix-stripping

**Comment by shuLhan** [comment]
> I think this is a bug. If the prefix never contained an absolute path before trying to strip it out, why would it start using an absolute path.

After some reading, testing and debugging; I think I know the reason why input path must be converted to absolute path: to handle recursive symlink of a sub directory to parent directory. I know it this is rare case on must user, but it is possible and must be handled by program.

The test case was in here [1]. To test whether relative path will work or not, try removing the if condition on this line [2], and you will get an error (infinite recursion) on function `findFiles`.

Assume that input directory was,
```
./D1
  + D2 -> symlink to D1
  + F1
```
To traverse the input directory D1, the program need to keep log of visited path. Let say that `./D1` was already visited, then we continue to D2. Since D2 is symlink to D1, when we read the real path we will get `../D1`. Now, do we have `../D1` in log of visited path? The answer is no, and the program will keep finding files recursively.

I hope that explain why I tag this as invalid.

[1] https://github.com/shuLhan/go-bindata/blob/73d211dfcb215f1fcd5694b48f1f03564ae09359/convert_test.go#L68
[2] https://github.com/shuLhan/go-bindata/blob/73d211dfcb215f1fcd5694b48f1f03564ae09359/convert.go#L73

**Comment by ghostsquad** [comment]
This makes sense, but it completely breaks the usability. Absolute paths
never make sense when accessing assets. So, although technically this might
mean somewhere in the logic, absolute paths need to be used, in the end,
should maintain the same relative path interface/expectations.

**Comment by shuLhan** [comment]
Hi @ghostsquad ,

I refactored the file scanner to use relative path instead of absolute path on branch dev-shulhan, not just to fix this issue, but for more code readibility. If you have time, can you try and test with your reported issue (and hopefully with your use case)?

Thank you.

**Comment by ghostsquad** [comment]
I've actually abandoned this project for https://github.com/shurcooL/vfsgen. It provides additional flexibility that I found I needed (mostly around creating virtual files).

**Comment by shuLhan** [comment]
Fixes by d6d5ca8.
